mlir/lib/Dialect/ArmSME/Transforms/LegalizeForLLVMExport.cpp

```diff
@@ -402,6 +402,48 @@ struct MoveVectorToTileSliceToArmSMELowering
   }
 };
 
+/// Lower `arm_sme.move_tile_slice_to_vector` to SME intrinsics. Only horizontal
+/// tile slices are currently supported.
+struct MoveTileSliceToVectorArmSMELowering
+    : public ConvertOpToLLVMPattern<arm_sme::MoveTileSliceToVectorOp> {
+  using ConvertOpToLLVMPattern<
+      arm_sme::MoveTileSliceToVectorOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(arm_sme::MoveTileSliceToVectorOp moveTileSliceToVector,
+                  OpAdaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = moveTileSliceToVector.getLoc();
+    auto sliceType = moveTileSliceToVector.getResult().getType();
```

Comment:
add a `getResultVectorType` method?
```suggestion
    auto sliceType = moveTileSliceToVector.getResultVectorType();
```

Reply:
Went with `getSliceType()`, since I think that's closer to the intent (and is similar to `getTileType()` on the `move_vector_to_tile_slice` op).